ml-agents/mlagents/trainers/torch/utils.py

```diff
@@ -115,57 +115,61 @@ def _check_resolution_for_encoder(
                 f"the provided EncoderType ({vis_encoder_type.value}). The min dimension is {min_res}"
             )
 
+    @staticmethod
+    def get_encoder_for_obs(
+        shape: Tuple[int, ...],
+        normalize: bool,
+        h_size: int,
+        vis_encode_type: EncoderType,
+    ) -> Tuple[nn.Module, int]:
+        """
+        Returns the encoder and the size of the appropriate encoder.
+        :param shape: Tuples that represent the observation dimension.
+        :param normalize: Normalize all vector inputs.
+        :param h_size: Number of hidden units per layer.
+        :param vis_encode_type: Type of visual encoder to use.
+        """
+        if len(shape) == 1:
```

Comment:
Eventually, this will use the DimensionProperties to decide an encoder correct?

Reply:
Yes